Source/WebCore/platform/network/BlobPart.h

```diff
@@ -30,58 +30,59 @@
 namespace WebCore {
 
 class BlobPart {
+private:
+    friend struct IPC::ArgumentCoder<BlobPart, void>;
 public:
     enum class Type : bool {
         Data,
         Blob
     };
 
     BlobPart()
-        : m_type(Type::Data)
+        : m_dataOrURL(Vector<uint8_t> { })
     {
     }
 
     BlobPart(Vector<uint8_t>&& data)
-        : m_type(Type::Data)
-        , m_data(WTFMove(data))
+        : m_dataOrURL(WTFMove(data))
     {
     }
 
     BlobPart(const URL& url)
-        : m_type(Type::Blob)
-        , m_url(url)
+        : m_dataOrURL(url)
     {
     }
 
-    Type type() const { return m_type; }
-
-    const Vector<uint8_t>& data() const
+    Type type() const
     {
-        ASSERT(m_type == Type::Data);
-        return m_data;
+        return std::holds_alternative<URL>(m_dataOrURL) ? Type::Blob : Type::Data;
     }
 
-    Vector<uint8_t> moveData()
+    Vector<uint8_t>&& moveData()
```

Comment:
Nit: Not sure the `&&` is really needed here

Reply:
There's a different between xvalues and prvalues - I once knew, but can't for the life of me remember them
I just know at least some of us got Into the habit of decorating for xvalue because it can help catch future coding mistakes (even if rare)